src/db/db-helper.js

```diff
@@ -579,4 +579,54 @@ module.exports = class DBHelper {
       throw err;
     }
   }
+  /* Names */
+  static async findName(query) {
+    try {
+      return db.Names.find(query);
+    } catch (err) {
+      logger.error(`FIND Names error: ${err.message}`);
+      throw err;
+    }
+  }
+
+  static async findOneName(query) {
+    try {
+      return db.Names.findOne(query);
+    } catch (err) {
+      logger.error(`FINDONE Names error: ${err.message}`);
+      throw err;
+    }
+  }
+
+  static async countName(query) {
+    try {
+      return db.Names.count(query);
+    } catch (err) {
+      logger.error(`COUNT Name error: ${err.message}`);
+      throw err;
+    }
+  }
+
+  static async insertName(nameEntry) {
+    try {
+      const existing = await DBHelper.findOneName({ address: nameEntry.address });
+      if (isNull(existing)) {
+        await db.Names.insert(nameEntry);
+      } else {
+        await DBHelper.updateName(nameEntry.address, nameEntry);
+      }
+    } catch (err) {
+      logger.error(`INSERT Name error: ${err.message}`);
+      throw err;
+    }
+  }
+
+  static async updateName(address, fields) {
```

Comment:
we dont need `update`. in the Naka ANS, you can only assign once and its permanent. so that means we have to be sure when we insert into the Nakabase DB that its good data.